src/store.rs

```diff
@@ -907,68 +615,89 @@ impl<F: LurkField> Store<F> {
         self.mark_dehydrated_cont(self.get_cont_dummy())
     }
 
-    pub fn scalar_from_parts(&self, tag: F, value: F) -> Option<ScalarPtr<F>> {
-        let Some(e_tag) = ExprTag::from_field(&tag) else { return None };
-        let scalar_ptr = ScalarPtr::from_parts(e_tag, value);
-        self.scalar_ptr_map
-            .contains_key(&scalar_ptr)
-            .then_some(scalar_ptr)
-    }
-
-    pub fn scalar_from_parts_cont(&self, tag: F, value: F) -> Option<ScalarContPtr<F>> {
-        let Some(e_tag) = ContTag::from_field(&tag) else { return None };
-        let scalar_ptr = ScalarContPtr::from_parts(e_tag, value);
-        if self.scalar_ptr_cont_map.contains_key(&scalar_ptr) {
-            return Some(scalar_ptr);
-        }
-        None
-    }
-
-    pub fn fetch_scalar(&self, scalar_ptr: &ScalarPtr<F>) -> Option<Ptr<F>> {
-        self.scalar_ptr_map.get(scalar_ptr).map(|p| *p)
+    pub fn fetch_scalar(&self, scalar_ptr: &ZExprPtr<F>) -> Option<Ptr<F>> {
```

Comment:
I know it's legacy, but this naming is a bit weird. I don't think it's a 'scalar' anything that's being fetched.